src/borg/helpers/passphrase.py

```diff
@@ -111,19 +115,27 @@ def verification(cls, passphrase):
         ):
             print('Your passphrase (between double-quotes): "%s"' % passphrase, file=sys.stderr)
             print("Make sure the passphrase displayed above is exactly what you wanted.", file=sys.stderr)
+            print(
+                "Your passphrase (UTF-8 encoding in hex): %s" % bin_to_hex(passphrase.encode("utf-8")), file=sys.stderr
+            )
             try:
                 passphrase.encode("ascii")
             except UnicodeEncodeError:
-                print(
-                    "Your passphrase (UTF-8 encoding in hex): %s" % bin_to_hex(passphrase.encode("utf-8")),
-                    file=sys.stderr,
-                )
                 print(
                     "As you have a non-ASCII passphrase, it is recommended to keep the "
                     "UTF-8 encoding in hex together with the passphrase at a safe place.",
                     file=sys.stderr,
                 )
 
+    @staticmethod
+    def display_debug_info(passphrase):
+        print(
+            "Incorrect passphrase (UTF-8 encoding in hex): %s" % bin_to_hex(passphrase.encode("utf-8")), file=sys.stderr
+        )
+        for env_var in ["BORG_PASSPHRASE", "BORG_PASSCOMMAND", "BORG_PASSPHRASE_FD"]:
+            env_var_value = os.environ.get(env_var)
+            print(f"{env_var} = {env_var_value}", file=sys.stderr)
```

Comment:
if would be good to output the env variables in a shell-like syntax (bash), including quotes around the value, so that leading/trailing blanks can be seen.

also guess you need to specialcase not set env vars, outputting "None" would be misleading.